subsys/net/lib/tftp/tftp_client.c

```diff
@@ -222,7 +222,11 @@ static int send_request(int sock, struct tftpc *client,
 		}
 
 		/* Limit communication to the specific address:port */
-		connect(sock, &from_addr, from_addr_len);
+		if (connect(sock, &from_addr, from_addr_len) < 0) {
+			LOG_ERR("connect failed, err %d", errno);
+			ret = -errno;
```

Comment:
It might make sense to switch the ordering of these statements as the errno might get set in logger. Also I am proposing that all the error values are printed as <0 just to make sure the value sinks in to the reader.
So suggesting
```
ret = -errno;
LOG_ERR("...", ret);
```

Reply:
That's good point, updated the commit as suggested.